packages/Filter/src/components/Item/Item.js

```diff
@@ -121,16 +125,27 @@ function Item(props) {
           <sc.RuleSelect data-pka-anchor="filter.item.ruleSelector">
             <ListBox onChange={handleChangeRule}>
               <ListBox.Trigger hasClearButton={false} />
-              {rulesByType[selectedColumnType].map(rule => (
-                <ListBox.Option key={rule} value={rule} isSelected={rule === selectedRule}>
-                  {I18n.t(`filter.rules.${localeKeysByRule[rule]}`)}
-                </ListBox.Option>
-              ))}
+              {rulesByType[selectedColumnType]
+                .filter(rule =>
+                  removeIllogicalRules(
+                    operator,
+                    rule,
+                    React.Children.toArray(children).map(child => child.props),
```

Comment:
If you move `React.Children.toArray(children).map(child => child.props),` outside, then this calculation will be run only once rather that on every step of the loop: 

```jsx
const existingFilters = React.Children.toArray(children).map(child => child.props);
```

-- 
Also, the fact that this calculation gives us `existingFilters` is not very obvious. But maybe that's alright.